compiler/passes/resolveIntents.cpp

```diff
@@ -47,6 +47,7 @@ static IntentTag constIntentForType(Type* t) {
              t == dtStringCopy ||
              t == dtCVoidPtr ||
              t == dtCFnPtr ||
+             // TODO: t->symbol->hasFlag(FLAG_RANGE) ||
```

Comment:
yes. I don't think we've decided yet what the default intent for ranges is. I think resolveIntents should be explicit about it instead of relying on ranges being "records".